### ISSUE TITLE
Functions are not successfully created in master (since update to Kubeless 0.6.0)

### ISSUE THREAD

**Issue by arapulido** [kubeapps/kubeapps#263]
Steps to reproduce:

* Using the dashboard, create a python function (example is fine)
* Refresh the Functions ui
* Click on the function

Results: You get a blank page

It looks like the function CRD object is create  successfully, but not the deployment

```
192 :: github.com/kubeapps/kubeapps ‹master*› » k get functions
NAME                      AGE
python-function-example   13m
rotten-toothbrush         14s
192 :: github.com/kubeapps/kubeapps ‹master*› » k get deploy
No resources found.
```

Errors in the kubeless-controller log:
```
time="2018-04-21T10:25:15Z" level=info msg="Processing change to Function default/rotten-toothbrush" pkg=controller
time="2018-04-21T10:25:16Z" level=error msg="Function can not be created/updated: Deployment.apps \"rotten-toothbrush\" is invalid: spec.template.spec.containers[0].image: Required value" pkg=controller
time="2018-04-21T10:25:16Z" level=error msg="Error processing default/rotten-toothbrush (will retry): Deployment.apps \"rotten-toothbrush\" is invalid: spec.template.spec.containers[0].image: Required value" pkg=controller
time="2018-04-21T10:25:16Z" level=info msg="Processing change to Function default/rotten-toothbrush" pkg=controller
time="2018-04-21T10:25:17Z" level=error msg="Function can not be created/updated: Deployment.apps \"rotten-toothbrush\" is invalid: spec.template.spec.containers[0].image: Required value" pkg=controller
time="2018-04-21T10:25:17Z" level=error msg="Error processing default/rotten-toothbrush (giving up): Deployment.apps \"rotten-toothbrush\" is invalid: spec.template.spec.containers[0].image: Required value" pkg=controller
E0421 10:25:17.856422       1 controller.go:181] Deployment.apps "rotten-toothbrush" is invalid: spec.template.spec.containers[0].image: Required value
```

**Comment by prydonius** [comment]
@arapulido can you verify that the kubeless-controller is v0.6.0? This might have happened because you were testing the feat/auth branch (and did `kubeapps up` with that) which is pointing to the older version.

**Comment by arapulido** [comment]
I have verified that this is running master. I have recreated my cluster, made sure I built kubeapps from master and did `kubeapps up` again. I still reproduce the issue.

**Comment by prydonius** [comment]
cc @andresmgot any ideas?

**Comment by sameersbn** [comment]
I can confirm this issue report. The kubeless-controller reports the following error in the logs:

```console
# kubectl logs -f kubeless-controller-manager-668c7878bd-qkmjh -n kubeless
time="2018-04-23T05:53:05Z" level=info msg="Starting kubeless controller" pkg=controller
time="2018-04-23T05:53:05Z" level=info msg="Kubeless controller synced and ready" pkg=controller
time="2018-04-23T05:56:08Z" level=info msg="Processing change to Function default/hello" pkg=controller
time="2018-04-23T05:56:09Z" level=error msg="Function can not be created/updated: Deployment.apps \"hello\" is invalid: spec.template.spec.containers[0].image: Required value" pkg=controller
time="2018-04-23T05:56:09Z" level=error msg="Error processing default/hello (will retry): Deployment.apps \"hello\" is invalid: spec.template.spec.containers[0].image: Required value" pkg=controller
```

**Comment by arapulido** [comment]
```
apiVersion: v1
items:
- apiVersion: kubeless.io/v1beta1
  kind: Function
  metadata:
    clusterName: ""
    creationTimestamp: 2018-04-21T16:30:20Z
    name: zealous-beast
    namespace: default
    resourceVersion: "701"
    selfLink: /apis/kubeless.io/v1beta1/namespaces/default/functions/zealous-beast
    uid: 48bd8e91-4581-11e8-83e6-080027dd980f
  spec:
    deps: ""
    function: |
      def handler():
        return "Hello World"
    handler: zealous-beast.handler
    runtime: python3.6
    type: HTTP
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

**Comment by prydonius** [comment]
@arapulido ah I see, you'll need to run telepresence, as the function template has changed: https://github.com/kubeapps/kubeapps/commit/463a82aaf8b1fe34abe9d24caa5ac5f1478d8f6e#diff-09a63a6e2ddfd0106b92aed2cfa32be6R227

**Comment by andresmgot** [comment]
yes, that function format is for Kubeless < 0.5. Should we update something? The latest `dashboard` should have the changes, isn't it?

**Comment by prydonius** [comment]
@andresmgot yes, but the manifest is deploying `kubeapps/dashboard:v0.6.0` which won't have the changes. It will when the new release comes out. Ideally we should push `latest` tags on master and have the manifests point to the `latest` tag, which would make testing master much easier.

**Comment by arapulido** [comment]
We should work on this. For me, this means that master is broken. Master should always work, without having to run telepresence. This is very confusing for people arriving to the project.

**Comment by andresmgot** [comment]
Agree, let me know if you want me to give a hand with that. Happy to help

**Comment by prydonius** [comment]
@arapulido agreed, should be fairly trivial to add latest tags to each master build, I'll look into it post-kubecon.

So after some investigation on this, it turns out we are pointing to a pretty old kubeless controller image (https://github.com/kubeapps/kubeapps/blob/master/manifests/kubeless.jsonnet#L5). I've fixed this in #267 so that we always use the tagged version from the vendored kubeless manifest.

**Comment by arapulido** [comment]
@prydonius I have filed https://github.com/kubeapps/kubeapps/issues/271
